pkg/volume/csi/csi_plugin.go

```diff
@@ -912,13 +912,11 @@ func (p *csiPlugin) getFSGroupPolicy(driver string) (storage.FSGroupPolicy, erro
 	}
 
 	// If the csiDriver isn't defined, return the default behavior
-	if csiDriver == nil {
+	// This also allows drivers that were created before the feature existed
+	// to use the default policy instead of throwing an error.
+	if csiDriver == nil || csiDriver.Spec.FSGroupPolicy == nil || *csiDriver.Spec.FSGroupPolicy == "" {
```

Comment:
Are we sure the apiserver got upgraded first before kubelet? API defaulting is supposed to be done by the apiserver when a client does a GET: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#defaulting

Reply:
For an upgraded API server, I verified that kubectl get csidriver call returns the default value "ReadWriteOnceWithFSType" for "fsGroupPolicy" field.

Reply:
even in case apiserver and kubelet are upgraded in the right order, this could happen if later the feature gate is disabled on apiserver but enabled on kubelet, since the defaulting will only happen if apiserver feature gate is enabled https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/pkg/apis/storage/v1/defaults.go#L56.

since nothing is stopping users from arriving at this 'invalid' feature gate configuration I guess kubelet should tolerate it.

Reply:
feature gate misconfiguration doesn't explain what happened in https://github.com/kubernetes/kubernetes/issues/99790 though.  They say they upgraded to 1.20 where the apiserver feature flag got turned on ..